### PR TITLE
gh-125911: Rename big trampoline to shim

### PR DESCRIPTION
**Pull request by savannahostrowski**
I _think_ I did this correctly/caught everything 😅. Basically, I renamed the `trampoline` associated with the StencilGroup (aka big trampoline) to `shim` and then left the rest as `trampolines` as those refer the little trampolines needed to compile a trace.

* Issue: gh-125911

### CODE REVIEW

File: Python/jit.c [hunk 1]
```diff
@@ -507,12 +507,12 @@ _PyJIT_Compile(_PyExecutorObject *executor, const _PyUOpInstruction trace[], siz
     unsigned char *code = memory;
     unsigned char *data = memory + code_size;
     state.trampolines.mem = memory + code_size + data_size;
-    // Compile the trampoline, which handles converting between the native
+    // Compile the shim, which handles converting between the native
     // calling convention and the calling convention used by jitted code
     // (which may be different for efficiency reasons). On platforms where
-    // we don't change calling conventions, the trampoline is empty and
+    // we don't change calling conventions, the shim is empty and
     // nothing is emitted here:
```
Comment (brandtbucher) [inline Python/jit.c]: I just realized that the comment "On platforms where we don't change calling conventions, the shim is empty and nothing is emitted here:" is outdated. Can you remove it?

File: Tools/jit/_writer.py [hunk 2]
```diff
@@ -19,14 +19,14 @@ def _dump_footer(
     yield "        const _PyUOpInstruction *instruction, jit_state *state);"
     yield "    size_t code_size;"
     yield "    size_t data_size;"
-    yield "    symbol_mask trampoline_mask;"
+    yield "    symbol_mask shim_mask;"
```
Comment (brandtbucher) [inline Tools/jit/_writer.py]: This member is indeed referring to the "little" trampolines:

```suggestion
    yield "    symbol_mask trampoline_mask;"
```

**Review by brandtbucher — COMMENTED** [review]
This is so much better, thanks. Just one incorrect rename... a testament to how confusing I made this, and how much this change is needed. ;)

@diegorusso: just a heads-up that we're renaming the "big" calling convention trampoline to `shim` and leaving all of your new "little" range-extending trampolines as `trampoline`.

File: Tools/jit/_writer.py (hunk 2 above)
Comment (savannahostrowski) [reply Tools/jit/_writer.py]: Ahh, you're right. Thanks!

**Comment by diegorusso** [comment]
Yes, I agree with the change. In fact I always called the "other trampoline" :) Thanks Savannah for the change, it LGTM. 🎉

**Review by brandtbucher — APPROVED** [review]

**Pull request merged**